services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,68 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (thermalMgrQueueHandle == NULL)
+    return ERR_CODE_INVALID_STATE;
 
+  if (event == NULL)
+    return ERR_CODE_INVALID_ARG;
+
+  if( xQueueSend(thermalMgrQueueHandle,
+                ( void * )event,
+                ( TickType_t ) 10) == errQUEUE_FULL)
+  {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  float tempC;
+  thermal_mgr_event_t eventCall;
+
+  error_code_t errTemp = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempC);
```

Comment:
you shouldn't be reading the temperature in an ISR
1. the temperature read uses I2C which is relatively slow and we want the ISR to be as quick as possible
2. this function call will eventually call one of the I2C helper functions, which will try to take the I2C bus mutex and block